components/TimeDisplay.js

```diff
@@ -0,0 +1,31 @@
+class TimeDisplay {
+  constructor(app) {
+    this.app = app;
+    this.timeElement = document.querySelector(".time");
+    this.dateElement = document.querySelector(".date");
+
+    this.setupCurrentTime();
+    this.setupCurrentDate();
+  }
+
+  setupCurrentTime() {
+    const currentDate = new Date();
+    const currentTimeString = currentDate.toLocaleTimeString().slice(0, 5);
+    this.timeElement.textContent = currentTimeString;
+  }
+
+  setupCurrentDate() {
+    const currentDate = new Date();
+    const currentDayOfMonth = currentDate.getDate();
+    const currentMonthString =
+      this.app.MONTHS_COLLECTION[currentDate.getMonth()];
+    const currentDayOfWeekString = this.app.DAY_OF_WEEK_COLLECTION[
+      currentDate.getDay()
+    ].slice(0, 3);
+    const currentDateString = `${currentDayOfWeekString} ${currentDayOfMonth} ${currentMonthString}`;
+
+    this.dateElement.textContent = currentDateString;
+  }
```

Comment:
move currentDate to remove duplication
```suggestion
    const currentDate = new Date();
    this.setTime(currentDate);
    this.setDate(currentDate);
  }

  setTime(currentDate) {
    const currentTimeString = currentDate.toLocaleTimeString().slice(0, 5);
    this.timeElement.textContent = currentTimeString;
  }

  setDate(currentDate) {
    const currentDayOfMonth = currentDate.getDate();
    const currentMonthString =
      this.app.MONTHS_COLLECTION[currentDate.getMonth()];
    const currentDayOfWeekString = this.app.DAY_OF_WEEK_COLLECTION[
      currentDate.getDay()
    ].slice(0, 3);
    const currentDateString = `${currentDayOfWeekString} ${currentDayOfMonth} ${currentMonthString}`;

    this.dateElement.textContent = currentDateString;
  }
```

Reply:
[8fa050b](https://github.com/VasiliiDubik/Vasiliidubik-weatherForecast/pull/1/commits/8fa050b992eee756600deeda3dcad963b942fb6a)